Notes_and_Examples/1_protractor_singleVector/1_protractor.xml

```diff
@@ -42,8 +35,15 @@ def vglcfn(e, ans):
           return 'The length of {} is incorrect. Your length: {:.1f}'.format(vec.name, vec.length)
   def check_angle(check, vectors):
       vec = vectors[check['vector']]
+      expected = check['expected']
       tolerance = check.get('tolerance', 2.0)
-      if abs(vec.angle - check['expected']) > tolerance:
+      upper_limit = (expected + tolerance) % 360
+      lower_limit = (expected - tolerance) % 360
+      if upper_limit < lower_limit:  # Can happen when upper_limit=2, lower_limit=-358, for example.
```

Comment:
@mtyaka comment is  not quite correct; -2 % 360 is `358`, not `-358`. Also, the overall approach is a bit weird. It works correctly under circumstances we have, but it is weird. Proper way to do the would be to calculate angle between target vector and identity vector with `expected` angle - which is (sin(expected), cos(expected)), but it involves scalar product and a bit involved.

This approach works, but it might make sense to slightly improve it to have more mathematical correctness. For example, if expected is 0 and tolerance is 3, 363 will pass the check, while it shouldn't. So you likely need to normalize (%360) vector angle and/or substitute check below with

    valid = 0 <= vec.angle <= upper_limit or lower_limit <= vec.angle <= 360